api/gear/historical_proxy.idl

```diff
@@ -0,0 +1,65 @@
+type Config = struct {
+  reply_timeout: u32,
+  reply_deposit: u64,
+};
+
+type ProxyError = enum {
+  NoEndpointForSlot: u64,
+  SendFailure: str,
+  ReplyTimeout: str,
+  ReplyFailure: str,
+  DecodeFailure: str,
+  NotAdmin,
+  ERC20Relay: Error,
+};
+
+type Error = enum {
+  DecodeReceiptEnvelopeFailure,
+  FailedEthTransaction,
+  AlreadyProcessed,
+  TooOldTransaction,
+  SendFailure,
+  ReplyFailure,
+  HandleResultDecodeFailure,
+  MissingCheckpoint,
+  InvalidBlockProof,
+  TrieDbFailure,
+  InvalidReceiptProof,
+  ReplyTimeout,
+  ReplyHook,
+  InvalidMessage,
+};
+
+constructor {
+  New : (config: Config);
+};
+
+service HistoricalProxy {
+  AddEndpoint : (slot: u64, endpoint: actor_id) -> result (null, ProxyError);
+  EndpointFor : (slot: u64) -> result (actor_id, ProxyError);
+  /// Redirect message to ERC20 Relay service which is valid for `slot`.
+  /// If message is relayed successfully then reply from relay service is sent to
+  /// `vft_gateway` address and proofs are returned.
+  /// 
+  /// # Parameters
+  /// 
+  /// - `slot`: slot for which message is relayed.
+  /// - `tx_index`: transaction index for message.
+  /// - `proofs`: SCALE encoded `EthToVaraEvent`.
+  /// - `client`: client address to send receipt to on success.
+  /// - `client_route`: route to send receipt to on success.
```

Comment:
To tell the truth I don't like that solution. From my point of view it's better to depend on `vft_gateway`/`manager` IDL since the file defines the interface (actually IDL stands for `Interface definition language`). And hence an actual instance of the interface is specified by the `client` address`.

I'm not insisting though. @mertwole it's up to you =)

Reply:
As I know `sails` don't allow to define **actual** interface that would be like `trait` in rust, so for now these IDL files contain something that's better to be called contract ABI. Actually I've proposed to define service separately and then inherit it in `vft-manager` and depend on it in `historical-proxy` but unfortunately there's no way of doing it